tgg/research/gnomad_tgps.py

```diff
@@ -0,0 +1,239 @@
+from gnomad_qc.v3.resources.release import release_sites
+from gnomad_qc.v3.resources.meta import meta
+import hailtop.fs as hfs
+import argparse
+import logging
+
+NUM_OF_GNOMAD_SAMPLES = 76156
+OUTPUT_FILENAME = "pangemoe_assessment_stats.txt"
+
+logging.basicConfig(
+    format="%(asctime)s (%(name)s %(lineno)s): %(message)s",
+    datefmt="%m/%d/%Y %I:%M:%S %p",
+)
+logger = logging.getLogger("Pangenome_assessment")
+logger.setLevel(logging.INFO)
+
+
+def get_pangenome_ids(
+        id_path: str,
+) -> hl.expr.expressions.typed_expressions.SetExpression:
+    """
+    Generate a Hail set of sample IDs in the pangenome.
+
+    :param str id_path: Cloud path to read in the .txt file containing pangenome sample
+    IDs.
+    :return: A set of sample IDs as hl.expr.expressions.typed_expressions.SetExpression.
+    """
+    ids = []
+    with hfs.open(id_path, 'r') as file:
+        for line in file:
+            sid = line.split(' ')[1].strip()
+            ids.append(sid)
+    ids = hl.set(ids)
+    return ids
+
+
+def write_output(
+        dat: hl.utils.struct.Struct,
+        file_dir: str,
+        filename: str,
+) -> None:
+    """
+    Write the output Struct containing computed stats to a .txt file.
+
+    :param hl.utils.struct.Struct dat: Data containing computed stats.
+    :param str file_dir: Output directory.
+    :param str filename: Output filename.
+    :return: None; function writes output to the output path.
+    """
+    with hfs.open(f'{file_dir}/{filename}', 'w') as f:
+        print(dat, file=f)
+
+
+def compute_stats(
+        pangenome_ids_path: str,
+        output_dir: str,
+) -> None:
+    """
+    Compute the stats of variants per gnomAD sample for gnomAD samples not in the
+    pangenome.
+
+    :param str pangenome_ids_path: Cloud path for the text file containing pangenome
+    sample IDs.
+    :param str output_dir: Cloud path to output the result file.
+    :return: None; function writes .txt file to output directory.
+    """
+    # Get variant site frequency table.
+    gnomad_freq_ht = release_sites().ht()
+
+    # Get gnomAD v3 data in split form.
+    mt = get_gnomad_v3_vds().variant_data
+    mt = hl.experimental.sparse_split_multi(mt)
+
+    # Get gnomAD v3 metadata and filter to released samples.
+    meta_ht = meta.ht()
+    meta_ht = meta_ht.filter(meta_ht.release == True)
+
+    # Use the filtered metadata to filter gnomAD v3 data.
+    mt = mt.filter_cols(hl.is_defined(meta_ht[mt.col_key]))
+    num_of_v3_samples = mt.count_cols()
+    logger.info("The number of samples in gnomAD v3 is: ", num_of_v3_samples)
+
+    # Get pangenome sample IDs.
+    pangenome_ids = get_pangenome_ids(pangenome_ids_path)
+    logger.info("The total number of samples in pangenome is ", hl.eval(hl.len(
+        pangenome_ids)))
+
+    # Filter mt to two new matrix tables: one containing gnomAD samples not in
+    # pangenome, and one containing samples in pangenome.
```

Comment:
Konrad had a good suggestion here of using this MT for the pangenome samples rather than using the raw v3 data: https://gnomad.broadinstitute.org/downloads#v3-hgdp-1kg (you'll want the sparse MT). this might be a resource in gnomad_qc, but I haven't checked.

that would make this code block look more like:
```
mt = mt.filter_cols(~pangenome_ids.contains(mt.s))
# Use the resource here instead of hardcoding this like I have
pg_mt = hl.read_matrix_table("gs://gcp-public-data--gnomad/release/3.1.2/mt/genomes/gnomad.genomes.v3.1.2.hgdp_1kg_subset_sparse.mt")
pg_mt = pg_mt.filter_cols(pangenome_ids.contains(pg_mt.s))
```